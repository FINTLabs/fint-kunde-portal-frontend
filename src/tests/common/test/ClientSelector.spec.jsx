import React from "react";
import { create } from "react-test-renderer";
import ClientSelector from "../../../common/test/ClientSelector";

describe("Client selector", () => {
  const component = create(
    <ClientSelector
      clients={[
        {
          dn: "cn=client1",
          shortDescription: "test1",
          assetId: "test.no"
        },
        {
          dn: "cn=client2",
          shortDescription: "test2",
          assetId: "test.no"
        }
      ]}
      handleChange={e => {
        let change = {};
        change[e.target.name] = e.target.value;
        this.setState(change);
      }}
      name="test"
      value="test"
    />
  );
  //const root = component.root;
  it("should match snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  /*
  it("changes value when selector changes", () => {
    root.props.handleChange;

    expect(true).toBe(true);
  });
  */
});
