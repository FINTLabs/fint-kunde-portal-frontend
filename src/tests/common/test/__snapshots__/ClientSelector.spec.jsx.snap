// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client selector should match snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="test"
    >
      Klient
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="select-test"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          
        </div>
        <input
          id="test"
          name="test"
          type="hidden"
          value="test"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <p
      className="MuiFormHelperText-root ClientSelector-clientWarning-1 MuiFormHelperText-filled"
    >
      Bruk en egen klient for testing. Passordet blir regenerert hver gang du kj√∏rer en test.
    </p>
  </div>
</div>
`;
